app/src/main/kotlin/cz/covid19cz/erouska/ui/mydata/MyDataVM.kt

```diff
@@ -33,128 +34,240 @@ class MyDataVM @ViewModelInject constructor(
     } else {
         SafeMutableLiveData(
             SimpleDateFormat(
                 LAST_UPDATE_UI_FORMAT,
                 Locale.getDefault()
-            ).format(Date(prefs.getLastStatsUpdate()))
+            ).format(Date(prefs.getLastMetricsUpdate()))
         )
     }
 
     fun getMeasuresUrl() = AppConfig.currentMeasuresUrl
 
     private fun getStats(date: String? = null) {
-
+        isLoading.value = true
         viewModelScope.launch {
             kotlin.runCatching {
                 return@runCatching firebaseFunctionsRepository.getStats(date)
             }.onSuccess { response ->
                 L.d(response.toString())
+                isLoading.value = false
 
-                safeLet(response.testsTotal, response.testsIncrease) { total, increase ->
+                safeLet(response.testsTotal,
+                    response.testsIncrease,
+                    response.testsIncreaseDate) { total, increase, increaseDate ->
                     testsTotal.value = total
                     testsIncrease.value = increase
 
                     prefs.setTestsTotal(total)
                     prefs.setTestsIncrease(increase)
+
+                    val lastUpdateDate = SimpleDateFormat(
+                        LAST_UPDATE_API_FORMAT,
+                        Locale.getDefault()
+                    ).parse(increaseDate)
+
+                    lastUpdateDate?.time?.let { lastUpdateMillis ->
+                        prefs.setTestsIncreaseDate(lastUpdateMillis)
+
+                        testsIncreaseDate.value = SimpleDateFormat(
+                            LAST_UPDATE_UI_FORMAT,
+                            Locale.getDefault()
+                        ).format(
+                            Date(lastUpdateMillis)
+                        )
+                    }
                 }
                 safeLet(
                     response.confirmedCasesTotal,
-                    response.confirmedCasesIncrease
-                ) { total, increase ->
+                    response.confirmedCasesIncrease,
+                    response.confirmedCasesIncreaseDate,
+                ) { total, increase, increaseDate ->
                     confirmedCasesTotal.value = total
                     confirmedCasesIncrease.value = increase
 
                     prefs.setConfirmedCasesTotal(total)
                     prefs.setConfirmedCasesIncrease(increase)
+
+                    val lastUpdateDate = SimpleDateFormat(
+                        LAST_UPDATE_API_FORMAT,
+                        Locale.getDefault()
+                    ).parse(increaseDate)
+
+                    lastUpdateDate?.time?.let { lastUpdateMillis ->
+                        prefs.setConfirmedCasesIncreaseDate(lastUpdateMillis)
+
+                        confirmedCasesIncreaseDate.value = SimpleDateFormat(
+                            LAST_UPDATE_UI_FORMAT,
+                            Locale.getDefault()
+                        ).format(
```

Comment:
NIT: Na formátování datumů jsem udělal extensiony Long.timestampToDate() atd., nešlo by to tady taky použít?